app/models/goal.py

```diff
@@ -1,5 +1,25 @@
-from sqlalchemy.orm import Mapped, mapped_column
+from sqlalchemy.orm import Mapped, mapped_column,relationship
 from ..db import db
+from sqlalchemy import String
+#from sqlalchemy import Date
+
 
 class Goal(db.Model):
+
     id: Mapped[int] = mapped_column(primary_key=True, autoincrement=True)
+    title: Mapped[str] = mapped_column(String(100), nullable=False)
+    tasks: Mapped[list["Task"]] = relationship(
+        back_populates="goal")
+
+    
+    def to_dict(self):
+        return {
+            "id": self.id,
+            "title": self.title
+        }
+
+    @classmethod
+    def from_dict(cls, data_dict):
+        if "title" not in data_dict:
+            raise KeyError("title")
```

Comment:
This pattern of checking for a key and raising a KeyError ourselves is an anti-pattern we should avoid. 

The Python language will raise a KeyError for us if we try to access the title key, so we should simplify our code by relying on that. 
- The create goal route already has a try/except to catch the raised error, so here in this function, we can remove this `if` block and let the error raise on the next line if it does come up. We can rely on the route to handle the raised error and use `abort` to send a correctly formed error response.